File: OMWAYServices/OMWAY/src/main/java/com/example/omway/dto/trip/RideDto.java
package com.example.omway.dto.trip;

import com.example.omway.model.trip.RideState;
import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;

import java.sql.Time;
import java.time.LocalDate;

@Data
public class RideDto {
    private Integer id;
    private Time pickUpTime;
    private Time dropOffTime;
    private String pickUpLocation;
    private String dropOffLocation;
    private Double distance;
    private LocalDate date;
    private String notes;
    private RideState state;
    private Double fare;
    private int rating;
    private String comment;
    private String riderId;
    private String driverId;
    private String carId;
}

File: OMWAYAppViews/OMWayApp/app/src/main/java/com/main/omwayapp/apirest/dto/trip/RideDto.kt
package com.main.omwayapp.apirest.dto.trip

import android.os.Parcelable
import com.google.gson.annotations.JsonAdapter
import kotlinx.parcelize.Parcelize
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import java.sql.Time
import java.time.LocalDate

@Parcelize
@Serializable
data class RideDto(
    val id:Int?,
    val pickUpTime: String,
    val dropOffTime: String?,
    val pickUpLocation:String,
    val dropOffLocation:String,
    val distance:Double,
    val date: String,
    val notes:String,
    val state: String,
    val fare: Double,
    val rating:Int?,
    val comment:String?,
    val riderId: String,
    val driverId:String?,
    val carId:String?
) : Parcelable
